utils/httpsignature/hmac.go

```diff
@@ -0,0 +1,39 @@
+package httpsignature
+
+import (
+	"crypto"
+	"crypto/hmac"
+	"crypto/sha512"
+	"crypto/subtle"
+	"io"
+)
+
+// HMACKey is a symmetric key that can be used for HMAC-SHA512 request signing and verification
+type HMACKey string
+
+// Sign the message using the hmac key
+func (key HMACKey) Sign(rand io.Reader, message []byte, opts crypto.SignerOpts) (signature []byte, err error) {
```

Comment:
Why is `rand` passed into this method, as well as the interface? There is no usage of it that I can see. Unless the intended interface was meant to abstract any type of signature that may require rand? Regardless my general recommendation is to never allow higher level functions to pass rand to the cryptographic algo due to api abuse (far too often callers passing rand are not sending enough - or any entropy).